cogs/rcon.py

```diff
@@ -0,0 +1,262 @@
+
+            #player info???
+            #eurules = get_nextmap('eu.tf2maps.net', 27015)
+            #await ctx.send(eurules)
+            #print(eurules)
+
+        elif server == "eu-mvm":
+            await ctx.send("eu mvm")
+        else:
+            await ctx.send("no server found")
+
+    #nextmap
+    @command()
+    @has_any_role('Staff', 'Server Mods', 'Senior Staff', 'Fub')
+    async def nextmap(self, ctx, server):
+        await ctx.trigger_typing()
+        if server == "us":
+
+            nextmap = get_nextmap('us.tf2maps.net', 27015)
+            await ctx.send(nextmap)
+
+        elif server == "us-mvm":
+
+            nextmap = get_nextmap('us.tf2maps.net', 27016)
+            await ctx.send(nextmap)
+
+        elif server == "eu":
+
+            nextmap = get_nextmap('eu.tf2maps.net', 27015)
+            await ctx.send(nextmap)
+
+        elif server == "eu-mvm":
+
+            nextmap = get_nextmap('eu.tf2maps.net', 27016)
+            await ctx.send(nextmap)
+        
+        else:
+            await ctx.send("no server found")
+
+#get server info via a2s
+def server_info(host, port):
+    return a2s.info((host, port))
+
+#get players via a2s
+def server_players(host, port):
+    return a2s.players((host, port))
+
+#get the next fucking map
+def get_nextmap(host, port):
+    rules = str(a2s.rules((host, port)))
+    
+    #turns a2s.rules into list
+    dalist = rules.split(',')
+
+    #jank ass code that grabs the sm_nextmap part of that sea of words
+    splitter = []
+    for x in dalist:
+        if "sm_nextmap" in x:
+            splitter.append(x)
+
+    #wish I didn't have to do this
+    almostthere = splitter[0]
+    #split the new list and grab the second half
+    nextmap = almostthere.split(':', 1)
+    #remove the dumb '
+    thenextmap = re.sub("'", ' ', nextmap[1])
+    
+    return thenextmap
+
+async def get_status(input):
+
+    #things we need to grab from status
+    #userid 1
+    #name 2
+    #uniqueid 3
+    #adr 8
+
+    #status header
+    header = input.split('#')[1]
+    print(header)
+    print("header")
+    
+    nospaceheader = re.sub(' +', ' ', header)
+    print(nospaceheader)
+
+    headersplit = nospaceheader.split(' ')
+    del headersplit[0]
+    headersplit.remove('connected')
+    headersplit.remove('ping')
+    headersplit.remove('loss')
+    headersplit.remove('state')
+
+    print(headersplit)
+
+    #drops top section before names on status
+    status = "```\n" + input.split('#', 1)[1] + "\n```"
+
+    #no excess spaces
+    nospacestatus = re.sub(' +', ' ', status)
+    print(nospacestatus)
+
+
+    splitboy = input.split('#', 2)[2]
+    nospace = re.sub(' +', ' ', splitboy)
+    print(str(nospace))
+
+    return status
+
+#ned to figure out how to do this to make less code
+async def run_rcon(server, command):
+
+    if server == "eu":
+        with Client(config.rcon.euip, port=config.rcon.imp, passwd=config.rcon.password) as client:
```

Comment:
we can refactor this to work based on keys that exist in the config. i can push a change for this